golem-worker-executor-base/src/services/scheduler.rs

```diff
@@ -135,6 +161,10 @@ impl SchedulerServiceDefault {
         let previous_hour_key = Self::schedule_key_from_timestamp(previous_hours_since_epoch);
         let current_hour_key = Self::schedule_key_from_timestamp(hours_since_epoch);
 
+        // TODO: couple of issues with this implementation
+        // 1: We only query scheduled actions for the current hour - 1. If we are unavailable for longer than that actions will not be run.
+        // 2: We use the timestamp of the scheduled action as a unique key. If we have 2 actions scheduled for the same point in time one will be silently discarded.
```

Comment:
I am not understanding. So can schedule by calling a host function. Now `n` number of workers running may schedule an invocation probably at the same time. Some of them happen to have exactly the same timestamp. And those will be discarded. Please help me understand, why there is a need of unique keys for scheduled actions?

Reply:
The problem is if for some reason we have multiple actions scheduled for the same point in time. We will overwrite those in the kv store -> only keeping one.